.github/workflows/cppcheck.yml

```diff
@@ -0,0 +1,23 @@
+name: Cppcheck Static Analysis
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '0 9 * * *' # Daily at 10:00 UTC
+jobs:
+  clang-analyzer:
```

Comment:
Could you change this name to cpp-analyzer or something, so it doesn't collide with the other "clang-analyzer" job